### PR TITLE
Add action for filtering FeatureData[AlphaDiversity] 

### PR DESCRIPTION
**Pull request by hagenjp**
changes to add functionality for filtering. Resolves issue #352

### CODE REVIEW

File: q2_diversity/_filter.py [hunk 1]
```diff
@@ -25,3 +27,18 @@ def filter_distance_matrix(distance_matrix: skbio.DistanceMatrix,
     except skbio.stats.distance.DissimilarityMatrixError:
         raise ValueError(
             "All samples were filtered out of the distance matrix.")
+
+
+# This function filters the SampleData[AlphaDiversity] table by the metadata
+# only sample present in the metadata will remain
```
Comment (colinvwood) [inline q2_diversity/_filter.py]: convert this to a docstring

File: q2_diversity/_filter.py [hunk 2]
```diff
@@ -25,3 +27,18 @@ def filter_distance_matrix(distance_matrix: skbio.DistanceMatrix,
...
+def filter_alpha_diversity_artifact(alpha_diversity: pd.Series,
+                                    metadata: qiime2.Metadata,
+                                    exclude_ids: bool = False) -> pd.Series:
+    ids_to_keep = metadata.get_ids()
+    if exclude_ids:
+        ids_to_keep = set(alpha_diversity.index) - set(ids_to_keep)
+    filtered_table = alpha_diversity[alpha_diversity.index.isin(ids_to_keep)]
```
Comment (colinvwood) [inline q2_diversity/_filter.py]: I don't think "table" is technically correct (`pd.Series` is one-dimensional), so maybe `filtered_metrics` or something

File: q2_diversity/tests/test_filter.py [hunk 3]
```diff
@@ -255,5 +256,68 @@ def test_with_exclude_ids_where_filter_all(self):
                                    exclude_ids=True)
 
 
+class TestFilterAlphaDiversityArtifact(unittest.TestCase):
```
Comment (colinvwood) [inline q2_diversity/tests/test_filter.py:259]: the only missing test case is `exclude_ids = True` and none filtered

File: q2_diversity/_filter.py [hunk 4]
```diff
@@ -25,3 +27,34 @@ def filter_distance_matrix(distance_matrix: skbio.DistanceMatrix,
...
+    """
+    This function filters the SampleData[AlphaDiversity] table by the metadata
+    only samples present in the metadata will remain.
+
+    Parameters
+    ----------
+    alpha_diversity : pd.Series
+    metadata : qiime2.Metadata
+    where : str, optional
+    exclude_ids : bool, optional
```
Comment (colinvwood) [inline q2_diversity/_filter.py]:
```suggestion
    alpha_diversity : pd.Series
        The alpha diversity metrics, indexed by sample.
    metadata : qiime2.Metadata
        The metadata object to be used for filtering.
    where : str, optional
        A SQLite WHERE clause specifying which samples to select from the
        metadata.
    exclude_ids : bool, optional
        Whether to keep (default) or exclude the selected IDs in the metadata.
```
